documentation/proposals/Proposal - Silk.NET 3.X Generic Maths.md

```diff
@@ -0,0 +1,884 @@
+    public static explicit operator Vector2D<float>(Vector2D<T> self);
+    public static explicit operator Vector2D<double>(Vector2D<T> self);
+    public static explicit operator Vector2D<decimal>(Vector2D<T> self);
+    public static explicit operator Vector2D<Complex>(Vector2D<T> self);
+    public static explicit operator Vector2D<BigInteger>(Vector2D<T> self);
+    public static explicit operator checked Vector2D<byte>(Vector2D<T> self);
+    public static explicit operator checked Vector2D<sbyte>(Vector2D<T> self);
+    public static explicit operator checked Vector2D<short>(Vector2D<T> self);
+    public static explicit operator checked Vector2D<ushort>(Vector2D<T> self);
+    public static explicit operator checked Vector2D<int>(Vector2D<T> self);
+    public static explicit operator checked Vector2D<uint>(Vector2D<T> self);
+    public static explicit operator checked Vector2D<long>(Vector2D<T> self);
+    public static explicit operator checked Vector2D<ulong>(Vector2D<T> self);
+    public static explicit operator checked Vector2D<Int128>(Vector2D<T> self);
+    public static explicit operator checked Vector2D<UInt128>(Vector2D<T> self);
+    public static explicit operator checked Vector2D<Half>(Vector2D<T> self);
+    public static explicit operator checked Vector2D<float>(Vector2D<T> self);
+    public static explicit operator checked Vector2D<double>(Vector2D<T> self);
+    public static explicit operator checked Vector2D<decimal>(Vector2D<T> self);
+    public static explicit operator checked Vector2D<Complex>(Vector2D<T> self);
+    public static explicit operator checked Vector2D<BigInteger>(Vector2D<T> self);
+
+    // Cast to System.Numerics.Vector2
+    public static explicit operator Vector2(Vector2D<T> self);
+    public static explicit operator checked Vector2(Vector2D<T> self);
+
+    // Downcast
+
+    // Upcast
+    public static explicit operator Vector3D<T>(Vector2D<T> self);
+    public static explicit operator Vector4D<T>(Vector2D<T> self);
+
+    // Upcast from System.Numerics.Vector < 2
+
+    // Downcast from System.Numerics.Vector >= 2
+    public static explicit operator Vector2D<T>(Vector2 self);
+    public static explicit operator checked Vector2D<T>(Vector2 self);
+    public static explicit operator Vector2D<T>(Vector3 self);
+    public static explicit operator checked Vector2D<T>(Vector3 self);
+    public static explicit operator Vector2D<T>(Vector4 self);
+    public static explicit operator checked Vector2D<T>(Vector4 self);
+    public static implicit operator Vector2D<T>((T X, T Y) components);
+    #endregion
+
+    public void Deconstruct(out T x, out T y);
+
+    /// <summary>Returns a vector whose elements are the absolute values of each of the specified vector's elements.</summary>
+    /// <param name="value">A vector.</param>
+    /// <returns>The absolute value vector.</returns>
+    static Vector2D<T> INumberBase<Vector2D<T>>.Abs(Vector2D<T> value);
```

Comment:
Why would `Abs` be explicit? It's an operation that works and has an intuitive answer for all types.

Reply:
There is another, non-explicit Abs method, and the implementation for this redirects to that. Thinking about it, though, it makes no sense to have both, so I'll adjust that momentarily.

Reply:
Update: This is actually here on purpose for INumberBase<T>, the intended implementation of Abs to be used is the one on the non-generic VectorXD.